### PR TITLE
dpdk: add warning in configuration phase and minor DPDK refactoring v2

### PR DESCRIPTION
**Pull request by lukashino**
Redmine ticket: 
https://redmine.openinfosecfoundation.org/issues/6739

v2 changes:
- change the format error message from `- (err %s)` to `: %s`

Describe changes:
- SCEnters added
- refactor code to smaller functions
- rewrite log message to be more consistent
- add a user warning on the change of RX/TX descriptors - on changes caused by the NIC adjustment

### CODE REVIEW

File: src/runmode-dpdk.c [hunk 1]
```diff
@@ -349,7 +349,7 @@ static void ConfigSetIface(DPDKIfaceConfig *iconf, const char *entry_str)
 
     retval = rte_eth_dev_get_port_by_name(entry_str, &iconf->port_id);
     if (retval < 0)
-        FatalError("Interface \"%s\": %s", entry_str, rte_strerror(-retval));
+        FatalError("Name of the interface (%s) is invalid: %s", entry_str, rte_strerror(-retval));
```
Comment (victorjulien) [inline src/runmode-dpdk.c:352]: this message suggests the name itself is invalid, but is that what is meant here? A name can be valid w/o being present in the system?

**Comment by suricata-qa** [comment]
Information:
 

ERROR: QA failed on SURI_TLPW2_autofp_suri_time.
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPW2_autofp_stats_chk | | |
| .uptime | 101 | 110 | 108.91% | 

Pipeline 18572

File: src/runmode-dpdk.c (hunk 1 above)
Comment (lukashino) [reply src/runmode-dpdk.c:352]: No, it was meant to say that the interface does not exist. Will reword.

**Comment by lukashino** [comment]
#10472 has a followup

**Pull request closed without merging**